core/src/apps/bitcoin/get_public_key.py

```diff
@@ -115,24 +117,39 @@ async def get_public_key(
         node=node_type,
         xpub=node_xpub,
         root_fingerprint=keychain.root_fingerprint(),
+        descriptor=descriptor,
     )
 
 
 def _xpub_descriptor(
-    node_xpub: str,
+    node: bip32.HDNode,
+    xpub_magic: int,
     address_n: list[int],
     script_type: InputScriptType,
     fingerprint: int,
-) -> str:
+) -> str | None:
     from trezor.enums import InputScriptType
 
     from apps.common import paths
 
     from .common import descriptor_checksum
 
-    if script_type != InputScriptType.SPENDTAPROOT:
-        raise ValueError("Unsupported script type.")
-    path = paths.address_n_to_str(address_n)
-    descriptor = f"tr([{fingerprint:08x}{path[1:]}]{node_xpub}/<0;1>/*)"
+    if script_type == InputScriptType.SPENDADDRESS:
+        fmt = "pkh({})"
+    elif script_type == InputScriptType.SPENDP2SHWITNESS:
+        fmt = "sh(wpkh({}))"
+    elif script_type == InputScriptType.SPENDWITNESS:
+        fmt = "wpkh({})"
+    elif script_type == InputScriptType.SPENDTAPROOT:
+        fmt = "tr({})"
+    else:
+        return None
```

Comment:
when does this happen?
or iow, why not raise an exception here like trezorlib did?

Reply:
There's `InputScriptType.SPENDMULTISIG` which is not currently handled, I guess the descriptor will look very different. There's also `InputScriptType.EXTERNAL` but I think that does not make sense in `get_public_key` context.

Reply:
i don't think `SPENDMULTISIG` makes sense either though? the `GetPublicKey` message does not have a `multisig` field and a "multisig public key at path" is not a thing that makes sense either way.

OTOH I see now that SPENDMULTISIG is handled explicitly in there so i guess no reason to just break someone's misconfigured setup...